base/applications/cmdutils/more/more.c

```diff
@@ -139,28 +361,112 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
      */
     ConClearLine(Pager->Screen->Stream);
 
+    switch (s_chSubCommand)
+    {
+        case L'P':
+            s_chSubCommand = 0;
+            Pager->ScrollRows = nLines;
+            return TRUE;
+        case L'S':
+            s_chSubCommand = 0;
+            s_dwFlags |= FLAG_PLUSn;
+            s_nNextLineNo = Pager->lineno + nLines;
+            Pager->ScrollRows = Pager->ScreenRows - 1;
+            return TRUE;
+        case L'-':
+            s_chSubCommand = 0;
+            Pager->PagerAction = MorePageActionDoNothing;
+            return TRUE;
+    }
+    s_chSubCommand = 0;
+
     /* Ctrl+C or Ctrl+Esc: Control Break */
-    if ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
-        ((KeyEvent.wVirtualKeyCode == L'C') &&
-         (KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED))))
+    if (fCtrl && ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
+                  (KeyEvent.wVirtualKeyCode == L'C')))
     {
         /* We break, output a newline */
         WCHAR ch = L'\n';
         ConStreamWrite(Pager->Screen->Stream, &ch, 1);
         return FALSE;
     }
 
-    /* 'Q': Quit */
-    // FIXME: Available only when command extensions are enabled.
-    if ((KeyEvent.wVirtualKeyCode == L'Q') &&
-        !(KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)))
+    if (fCtrl)
     {
-        /* We break, output a newline */
-        WCHAR ch = L'\n';
-        ConStreamWrite(Pager->Screen->Stream, &ch, 1);
-        return FALSE;
+        /* Do nothing */
+        Pager->PagerAction = MorePageActionDoNothing;
+        return TRUE;
+    }
```

Comment:
I'm not convinced by the pertinence of this part of code. The original idea of the PagePrompt() function was that it does all the prompt management and there is no need at all to quit that function and tell the paging support to "not do anything".

Reply:
Deleted pager actions. OK.